src/configs/websocket/index.ts

```diff
@@ -62,14 +69,28 @@ export class ApiManager {
     if (this.pingInterval) {
       clearInterval(this.pingInterval);
     }
+    if (this.reconnectInterval) {
+      clearInterval(this.reconnectInterval);
+    }
     this.socket.addEventListener('open', () => {
+      this.websocket_connected && this.websocket_connected(true);
       this.pingInterval = setInterval(() => {
         this.socket.send(JSON.stringify({ ping: 1 }));
       }, PING_INTERVAL);
     });
 
     this.socket.addEventListener('close', () => {
+      this.websocket_connected && this.websocket_connected(false);
+      this.websocket_authorize && this.websocket_authorize(false);
       clearInterval(this.pingInterval);
+      this.socket = null;
+      if (attempts > 0) {
+        this.reconnectInterval = setTimeout(this.init.bind(this), RECONNECT_INTERVAL);
+        attempts -= 1;
+      } else {
+        window.alert('server down!!!');
```

Comment:
I don't suggest having an alert but implementing an in-house error handler component.
Alert is blocking the execution of the rest of the code and not a good user experience!

Reply:
Yes Matin, i have already dropped a message on the squad channel regarding as the requirement is not really clear on this. If we need to display message then we will need a design for it. I added this window alert as a temporary replacement to check the flow.